File: app-mvvm/src/main/java/com/xuie/imaginaryandroid/gui/nets/detail/NetsOnePresenter.java
package com.xuie.imaginaryandroid.gui.nets.detail;

import com.xuie.imaginaryandroid.data.source.NetsRepository;

import io.reactivex.android.schedulers.AndroidSchedulers;

import static com.xuie.imaginaryandroid.util.Utils.checkNotNull;

/**
 * @author xuie
 * @date 17-7-5
 */
public class NetsOnePresenter implements NetsOneContract.Presenter {
    private NetsRepository netsRepository;
    private NetsOneContract.View newsOneView;
    private int currentPage = 1;

    public NetsOnePresenter(NetsRepository netsRepository, NetsOneContract.View newsOneView) {
        this.netsRepository = checkNotNull(netsRepository);
        this.newsOneView = checkNotNull(newsOneView);
    }

    @Override
    public void subscribe() {
    }

    @Override
    public void unsubscribe() {

    }

    @Override
    public void getNewsOneRequest(String postId) {
        netsRepository.getNewDetail(postId)
                .observeOn(AndroidSchedulers.mainThread())
                .subscribe(netsDetail -> newsOneView.refreshNewsOne(netsDetail),
                        Throwable::printStackTrace);
    }
}

File: app-mvp-kotlin/src/main/java/com/xuie/imaginaryandroid/gui/nets/detail/NetsOnePresenter.kt
package com.xuie.imaginaryandroid.gui.nets.detail

import com.xuie.imaginaryandroid.data.source.NETSRepository

import io.reactivex.android.schedulers.AndroidSchedulers

/**
 * Created by xuie on 17-7-5.
 */

class NetsOnePresenter(
        private val netsRepository: NETSRepository,
        private val newsOneView: NetsOneContract.View
) : NetsOneContract.Presenter {
    private val currentPage = 1

    init {
        this.newsOneView.presenter = this
    }

    override fun subscribe() {
    }

    override fun unsubscribe() {

    }

    override fun getNewsOneRequest(postId: String) {
        netsRepository.getNewDetail(postId)
                .observeOn(AndroidSchedulers.mainThread())
                .subscribe({ netsDetail -> newsOneView.refreshNewsOne(netsDetail) },
                        { e -> e.printStackTrace()})
    }
}
